pandas/tests/io/json/test_pandas.py

```diff
@@ -1671,3 +1671,11 @@ def test_to_s3(self, s3_resource):
         assert target_file in (
             obj.key for obj in s3_resource.Bucket("pandas-test").objects.all()
         )
+
+    @pytest.mark.parametrize(
+        "dataframe,expected", [(pd.DataFrame([[pd.NA]]), '{"0":{"0":null}}',)],
```

Comment:
Can you add pd.NA to the indices of the DataFrame? There are different code paths for encoding values vs index labels so want to cover both 

You can see NdArray_encodeLabels for how that works for labels

Reply:
Note that we also have a `nulls_fixture` in the top level conftest that I think we could use here and probably just add pd.NA to @jorisvandenbossche

Reply:
I've added a test for `pd.NA` as an index label.  It currently encodes `pd.NA` as `"<NA>"` as it uses `NAType.__repr__`.  For reference, `np.nan` encodes as `"nan"` and `None` encodes as `"None"` when they are index labels.  Is this the desired behavior?

I'll have performance test result comparisons by tomorrow.

Reply:
> Note that we also have a nulls_fixture in the top level conftest that I think we could use here and probably just add pd.NA to @jorisvandenbossche

That might not be that easy, as `nulls_fixture` is used in many places, which not necessarily all support pd.NA 
(but it would be an interesting exercise to do, but not necessarily in this PR)

> It currently encodes pd.NA as "<NA>" as it uses NAType.__repr__. For reference, np.nan encodes as "nan" and None encodes as "None" when they are index labels. Is this the desired behavior?

I am not sure this is that useful, but at least it is consistent with the other null values.

Reply:
I think it is a "regression" for those in the index labels to not write "null". This was fixed for pd.NaT in #30977 but not the rest

I say regression in quotes because I don't think the behavior is well defined, but in any case a separate issue

Reply:
So for the index, this should also give "null" then?

Reply:
Yea I think so. AFAIK there is no JSON standard for how to represent missing values as object keys (or if you even should), but object keys MUST be strings, so `"null"` is probably the most reasonable accommodation. 

I think in 0.25 these probably just showed as `null` as an object key, which I don't think would be standard JSON anyway